src/app/services/token-lifetime.service.ts

```diff
@@ -25,6 +29,21 @@ export class TokenLifetimeService {
   protected terminateCancelTimeout: Timeout;
   private resumeBound;
 
+  // Check if token was used no more than 15 minutes ago
+  get isTokenWithinTimeline(): boolean {
+    const tokenLastUsed = this.window.localStorage.getItem('tokenLastUsed');
+
+    if (!tokenLastUsed) {
+      return false;
+    }
+
+    const tokenLastUsedDate = new Date(tokenLastUsed).getTime();
+    const tokenLifetime = 60 * 15 * 1000;
```

Comment:
> Shouldn't we rely on a system-configured token lifetime?

Token `system-configured token lifetime` is only known for the logged in user.
We are dealing with the possibly logged in user, but we wan't to avoid calling middleware with invalid token.
But I agree that we should change the value from 15min to 5min as the default min middleware value for token lifetime!